internal/pkg/term/selector/creds.go

```diff
@@ -0,0 +1,131 @@
+)
+
+const (
+	tempCredsOption       = "Enter temporary credentials"
+	accessKeyIDPrompt     = "What's your AWS Access Key ID?"
+	secretAccessKeyPrompt = "What's your AWS Secret Access Key?"
+	sessionTokenPrompt    = "What's your AWS Session Token?"
+)
+
+// Names wraps the method that returns a list of names.
+type Names interface {
+	Names() []string
+}
+
+// SessionProvider wraps the methods to create AWS sessions.
+type SessionProvider interface {
+	Default() (*session.Session, error)
+	FromProfile(name string) (*session.Session, error)
+	FromStaticCreds(accessKeyID, secretAccessKey, sessionToken string) (*session.Session, error)
+}
+
+// CredsSelect prompts users for credentials.
+type CredsSelect struct {
+	Prompt  Prompter
+	Profile Names
+	Session SessionProvider
+}
+
+// Creds prompts users to choose either use temporary credentials or choose from one of their existing AWS named profiles.
+func (s *CredsSelect) Creds(prompt, help string) (*session.Session, error) {
+	profileFrom := make(map[string]string)
+	options := []string{tempCredsOption}
+	for _, name := range s.Profile.Names() {
+		pretty := fmt.Sprintf("[profile %s]", name)
+		options = append(options, pretty)
+		profileFrom[pretty] = name
+	}
+
+	selected, err := s.Prompt.SelectOne(
+		prompt,
+		help,
+		options)
+	if err != nil {
+		return nil, fmt.Errorf("select credential source: %w", err)
+	}
+
+	if selected == tempCredsOption {
+		return s.askTempCreds()
+	}
+	sess, err := s.Session.FromProfile(profileFrom[selected])
+	if err != nil {
+		return nil, fmt.Errorf("create session from profile %s: %w", profileFrom[selected], err)
+	}
+	return sess, nil
+}
+
+func (s *CredsSelect) askTempCreds() (*session.Session, error) {
+	defaultAccessKey, defaultSecretAccessKey, defaultSessToken := s.defaultCreds()
+
+	accessKeyID, err := s.askWithMaskedDefault(accessKeyIDPrompt, defaultAccessKey)
+	if err != nil {
+		return nil, fmt.Errorf("get access key id: %w", err)
+	}
+	secretAccessKey, err := s.askWithMaskedDefault(secretAccessKeyPrompt, defaultSecretAccessKey)
+	if err != nil {
+		return nil, fmt.Errorf("get secret access key: %w", err)
+	}
+	sessionToken, err := s.askWithMaskedDefault(sessionTokenPrompt, defaultSessToken)
+	if err != nil {
+		return nil, fmt.Errorf("get session token: %w", err)
+	}
+
+	sess, err := s.Session.FromStaticCreds(accessKeyID, secretAccessKey, sessionToken)
+	if err != nil {
+		return nil, fmt.Errorf("create session from temporary credentials: %w", err)
+	}
+	return sess, nil
+}
+
+func (s *CredsSelect) askWithMaskedDefault(msg, defaultValue string) (string, error) {
+	accessKeyId, err := s.Prompt.Get(msg, "", nil, prompt.WithDefaultInput(mask(defaultValue)))
+	if err != nil {
+		return "", err
+	}
+	if accessKeyId == mask(defaultValue) {
+		// Return the original default instead of the masked value.
+		return defaultValue, nil
+	}
+	return accessKeyId, nil
+}
+
+func (s *CredsSelect) defaultCreds() (accessKeyID, secretAccessKey, sessionToken string) {
```

Comment:
Feel like this method should not be placed under `CredsSelect`... since it doesn't consume any of it's method/field value. What do you think of moving it to `sessions` pkg?

Reply:
That's a good idea!

Reply:
Actually, the behavior within the method is pretty unusual 
-> Try to get credentials, but if an error occurs leave them empty.

So I kept it in this package but instead of having it as a method, i changed it to a function so that it's  not namedspaced only to the `CredsSelect` struct